controllers/secret.go

```diff
@@ -24,187 +24,108 @@ import (
-	if value, ok := gitHubSecret.Data[GITHUB_TOKEN]; ok {
-		if value == nil || string(value) == "" {
-			return errors.New(fmt.Sprintf("github api secret found, but has an empty token value for: %s", GITHUB_TOKEN))
-		}
-
-		log.Info(fmt.Sprintf("github api secret with proper token name found in namespace: %s", meta.Namespace))
-	} else {
-		return errors.New(fmt.Sprintf("github api secret found, but has an improper token name, token name should be: %s", GITHUB_TOKEN))
-	}
-
-	return nil // Existing Secret found, do nothing...
+	return r.ensureSecret(secretName, GITHUB_API_KEY_NAME, meta)
 }
 
 // reconcilePyxisAPISecret will ensure that the Pyxis API Secret is present and up to date.
-func (r *OperatorPipelineReconciler) reconcilePyxisAPISecret(meta metav1.ObjectMeta) error {
-	pipelineName := types.NamespacedName{
-		Namespace: meta.Namespace,
-		Name:      meta.Name,
-	}
-	pipeline := &certv1alpha1.OperatorPipeline{}
-	err := r.Client.Get(context.TODO(), pipelineName, pipeline)
+func (r *OperatorPipelineReconciler) ensurePyxisAPISecret(meta metav1.ObjectMeta) error {
+	operatorPipeline, err := r.getPipeline(meta)
 	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			log.Info("pipeline resource not found. Ignoring since object must be deleted")
-			return nil
-		}
-		log.Error(err, "unable to retrieve pipeline resource in namespace "+meta.Namespace)
+		log.Error(err, "unable to resolve pyxis secret for %s in %s", meta.Name, meta.Namespace)
 		return err
 	}
-	var secretName = PYXIS_API_SECRET
-	if pipeline.Spec.PyxisApiSecretName != "" {
-		secretName = pipeline.Spec.PyxisApiSecretName
-	}
-	key := types.NamespacedName{
-		Namespace: meta.Namespace,
-		Name:      secretName,
+	secretName := PYXIS_API_SECRET_NAME
+	if operatorPipeline.Spec.PyxisSecretName != "" {
+		secretName = operatorPipeline.Spec.PyxisSecretName
 	}
-	secret := newSecret(key)
-	if !IsObjectFound(r.Client, key, secret) {
+	return r.ensureSecret(secretName, PYXIS_API_KEY_NAME, meta)
+}
+
+func (r *OperatorPipelineReconciler) ensureSecret(secretName string, secretKeyName string, meta metav1.ObjectMeta) error {
+	namespacedSecretName := newNamespacedName(secretName, meta.Namespace)
+	secret := &corev1.Secret{}
+	if !IsObjectFound(r.Client, namespacedSecretName, secret) {
 		err := errors.New(fmt.Sprintf("an existing secret named %s was not found in namespace %s", secretName, meta.Namespace))
 		log.Error(err, fmt.Sprintf("unable to reconcile %s in namespace %s", secretName, meta.Namespace))
 		return err
 	}
-	log.Info(fmt.Sprintf("existing %s secret found in namespace %s", secretName, meta.Namespace))
-	pyxisApiSecret := &corev1.Secret{}
-	if r.Client.Get(context.TODO(), key, pyxisApiSecret) != nil {
-		log.Error(err, fmt.Sprintf("unable to get the %s secret", secretName))
+	log.Info(fmt.Sprintf("an existing secret named %s was found in namespace %s", secretName, meta.Namespace))
+	secret = &corev1.Secret{}
```

Comment:
Has anything changed from line 84? Meaning do we need to do this reassignment?

Reply:
nope you are right its redundant